galasa-extensions-parent/dev.galasa.ras.couchdb/src/main/java/dev/galasa/ras/couchdb/internal/CouchdbRunResult.java

```diff
@@ -7,25 +7,32 @@
 
 import java.nio.file.Path;
 
+import dev.galasa.extensions.common.api.LogFactory;
 import dev.galasa.framework.spi.IRunResult;
 import dev.galasa.framework.spi.ResultArchiveStoreException;
+import dev.galasa.framework.spi.ras.ResultArchiveStoreFileStore;
 import dev.galasa.framework.spi.teststructure.TestStructure;
 import dev.galasa.ras.couchdb.internal.pojos.TestStructureCouchdb;
 
 public class CouchdbRunResult implements IRunResult {
 
     private final TestStructureCouchdb   testStructure;
-    private final Path            path;
     private final CouchdbRasStore store;
+    private final CouchdbDirectoryService storeService;
+    private Path path;
 
-    public CouchdbRunResult(CouchdbRasStore store, TestStructureCouchdb testStructure, Path path) {
+    public CouchdbRunResult(CouchdbRasStore store, TestStructureCouchdb testStructure, LogFactory logFactory) {
         this.store = store;
+        this.storeService = (CouchdbDirectoryService) store.getDirectoryServices().get(0);
```

Comment:
How do we know the first directory service is the right one to use?

Reply:
No idea. Ideally each service implementation would provide metadata in this...

> The OSGi Metatype specification provides a unified way to describe metadata about services and a Java interface to retrieve and manipulate such information. This information can optionally be localized, thus allowing the developer to provide human-readable metadata descriptions in multiple languages

or something. In our case, the only store is couchdb. And the cast would blow up if that wasn't the case...

Far from ideal that the outside code even knows that it's couchdb storing the data if you ask me.